tests/utils/test_sampling_algorithms.py

```diff
@@ -110,3 +118,79 @@ def test_farthest_point_sampling_pathological_case():
     with pytest.warns(UserWarning, match="identical"):
         selection = farthest_point_sampling(points, 2)
     assert selection == [0, 1]
+
+
+@pytest.mark.skipif(
+    not FPSAMPLE_INSTALLED, reason="Optional fpsample dependency not installed."
+)
+def test_fps_recommender_calls_fpsample(searchspace):
+    """Test that FPSRecommender uses fpsample when it is available."""
+    with patch("baybe._optional.fpsample.fps_sampling") as mock_fps:
+        mock_fps.return_value = np.array([0, 1, 2])
+
+        recommender = FPSRecommender()
+        result = recommender.recommend(3, searchspace)
+
+        mock_fps.assert_called_once()
+        assert result.index.tolist() == [0, 1, 2]
+
+
+@pytest.mark.skipif(
+    not FPSAMPLE_INSTALLED, reason="Optional fpsample dependency not installed."
+)
+@pytest.mark.parametrize(
+    ("init", "tie_break", "match"),
+    [
+        param(FPSInitialization.RANDOM, True, "initialization=", id="warn_initial"),
+        param(FPSInitialization.FARTHEST, True, "random tie-breaking", id="tiebreak"),
+    ],
+)
+def test_fps_recommender_warns_ignored_arguments(searchspace, init, tie_break, match):
+    """Test that FPSRecommender emits warnings for each unsupported argument."""
+    recommender = FPSRecommender(initialization=init, random_tie_break=tie_break)
+
+    with pytest.warns(UserWarning, match=match):
+        recommender.recommend(batch_size=3, searchspace=searchspace)
+
+
+def test_fps_recommender_fallback_to_internal_fps(searchspace):
+    """Test FPSRecommender will revert to internalFPS when `fpsample` unavailable."""
+    recommender = FPSRecommender()
+
+    with patch(
+        "baybe._optional.fpsample.fps_sampling",
+        side_effect=OptionalImportError("fpsample", "sampling"),
+    ):
+        with patch(
+            "baybe.recommenders.pure.nonpredictive.sampling.farthest_point_sampling"
+        ) as mock_internal:
+            mock_internal.return_value = [0, 1, 2]
+
+            result = recommender.recommend(
+                batch_size=3,
+                searchspace=searchspace,
+            )
+
+            mock_internal.assert_called_once()
+            assert result.index.tolist() == [0, 1, 2]
+
+
+@pytest.mark.skipif(
+    not FPSAMPLE_INSTALLED, reason="Optional fpsample dependency not installed."
+)
+def test_fps_recommender_known_grid_indices(searchspace):
```

Comment:
Could we make this test with all possible fps packages that may be used - test in both fps implementation modes